cli/azd/pkg/tools/git/git.go

```diff
@@ -245,3 +248,58 @@ func (cli *gitCli) IsUntrackedFile(ctx context.Context, repositoryPath string, f
 
 	return false, nil
 }
+
+// SetGitHubAuthForRepo creates git config for the repositoryPath like
+//
+// [credential "https://github.com"]  (when credential is equal to "https://github.com")
+// helper =
+// helper = !ghPath auth git-credential
+//
+// This way, git commands run from repositoryPath will use gh as auth credential.
+// Note: Removes any previous configuration for the credential.
+// Note: `helper = ` is intentional to break the chain of previously configured global helpers.
+// See: https://github.com/cli/cli/issues/3796 for more about this strategy.
+func (cli *gitCli) SetGitHubAuthForRepo(ctx context.Context, repositoryPath, credential, ghPath string) error {
+
+	if err := setAuthCredentialHelper(
+		ctx, cli.commandRunner, repositoryPath, credential, "", "replace-all"); err != nil {
+		return err
+	}
+	// path needs to be quoted on windows
+	if runtime.GOOS == "windows" {
+		ghPath = fmt.Sprintf("'%s'", ghPath)
+	}
+	ghCredentialValue := fmt.Sprintf("!%s auth git-credential", ghPath)
+	if err := setAuthCredentialHelper(
+		ctx, cli.commandRunner, repositoryPath, credential, ghCredentialValue, "add"); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func setAuthCredentialHelper(
+	ctx context.Context, runner exec.CommandRunner, repositoryPath, credential, value, flag string) error {
+	runArgs := newRunArgs(
+		"-C", repositoryPath,
+		"config", fmt.Sprintf("--%s", flag),
```

Comment:
Consider passing `--local` explicitly (I know it is the default, but it might be a nice "belt and suspenders" thing, and also make it clear what our intention was here)